inc/namespace.php

```diff
@@ -561,10 +567,19 @@ function set_tachyon_hostname( string $tachyon_url ) : string {
 /**
  * Ensure the S3 Uploads Bucket URL matches the current site hostname.
  *
+ * Because this filter involves switching blogs, the results are cached in a
+ * static variable for performance.
+ *
  * @param array $dirs Uploads directories array.
  * @return array
  */
 function set_s3_uploads_bucket_url_hostname( array $dirs ) : array {
+	static $cached_upload_dirs = null;
```

Comment:
The `= null` bit is redundant here as that's the value it'll have anyway before being assigned. I think we can make the implementation consistent between the 2 functions here and leave off the default values.
```suggestion
	static $cached_upload_dirs;
```